cranelift/codegen/src/isa/riscv64/inst.isle

```diff
@@ -1553,10 +1551,12 @@
   (if-let (i64_generate_imm imm20 imm12) (i64_sextend_u64 ty c))
   (rv_addi (rv_lui imm20) imm12))
 
-;; Otherwise we fall back to loading the immediate from memory
-;; TODO: This doesen't yet emit the constant into the memory pool, but it would
-;; be pretty neat if it did.
-(rule 0 (imm (ty_int ty) c) (emit_load_const_64 c))
+;; Otherwise we fall back to loading the immediate from the constant pool.
+(rule 0 (imm (ty_int ty) c)
+  (emit_load
+    (LoadOP.Ld)
+    (mem_flags_trusted)
+    (gen_const_amode (emit_u64_le_const c))))
```

Comment:
This is preexisting I think, and probably affects other backends too, but technically emitting a 64-bit constant here is over-approximating how large it needs to be, right? In that only the `ty`-width sized bit-pattern of `c` is all that matters so for a 32-bit constant the 64-bit size is overkill?

No need to fix here since this preserves what was done before, but may be worth trying to fix later?

Reply:
My plan is to eventually fix #6922 and then we should be able to change this to only ever trigger for 64 bit constants, since all other constants should be able to be emitted with the previous rules (i.e. with constant materialization instructions).